server/src/main/java/org/apache/gravitino/server/web/rest/ModelOperations.java

```diff
@@ -92,6 +98,15 @@ public Response listModels(
           () -> {
             NameIdentifier[] modelIds = modelDispatcher.listModels(modelNs);
             modelIds = modelIds == null ? new NameIdentifier[0] : modelIds;
+            modelIds =
+                MetadataFilterHelper.filterByExpression(
+                    metalake,
+                    "ANY(OWNER,METALAKE,CATALOG,SCHEMA) || "
```

Comment:
Schema owner don't have `use catalog` privilege. You should have `use catalog` privilege first.